extensions/photoroom-image-editing/index.tsx

```diff
@@ -0,0 +1,393 @@
+        canChooseDirectories
+        {...itemProps.imagePaths}
+      />
+
+      <Form.Separator />
+
+      <Form.Description text="Background" />
+      <Form.Checkbox label="Remove background" storeValue {...itemProps.removeBackground} />
+      <Form.TextField
+        title="Background color"
+        placeholder="6 digit color hex"
+        info="E.g., #000000 (fully transparent) or #ffffff (white)"
+        {...itemProps.backgroundColor}
+      />
+      <Form.Description text="OR" />
+      <Form.TextArea
+        title="Background prompt"
+        placeholder="Enter a prompt"
+        info="Prompt to use for guiding the background generation process."
+        {...itemProps.backgroundPrompt}
+      />
+
+      <Form.Separator />
+
+      <Form.Description text="AI Shadow" />
+      <Form.Dropdown
+        title="Add shadow"
+        storeValue
+        info="Shadow generation mode to use on the subject."
+        {...itemProps.shadowMode}
+      >
+        <Form.Dropdown.Item value="none" title="None" />
+        <Form.Dropdown.Item value="ai.soft" title="Soft" />
+        <Form.Dropdown.Item value="ai.hard" title="Hard" />
+        <Form.Dropdown.Item value="ai.floating" title="Floating" />
+      </Form.Dropdown>
+
+      <Form.Separator />
+
+      <Form.Description text="Position" />
+      <Form.Dropdown
+        title="Scaling Mode"
+        info="Whether the subject should fit (default) or fill the output image. If set to fit, the empty pixels will be transparent."
+        {...itemProps.scalingMode}
+      >
+        <Form.Dropdown.Item value="fit" title="Fit" />
+        <Form.Dropdown.Item value="fill" title="Fill" />
+      </Form.Dropdown>
+      <Form.Checkbox
+        label="Snapped to cropped edges"
+        info="If enabled (default), cropped sides of the subject will snap to the edges. For instance, for a portrait image cropped below the elbows, the subject will be aligned at the bottom even if a bottom padding is provided (but it will still respect bottom margin). Can't be set if background removal is disabled."
+        {...itemProps.snappedToCroppedEdges}
+      />
+
+      <Form.Separator />
+
+      <Form.Description text="Size" />
+      <Form.Dropdown
+        title="Output Size"
+        info="The size of the output image. If set to original image, the original size of the image will be used."
+        {...itemProps.outputSize}
+      >
+        <Form.Dropdown.Item value="custom" title="Custom" />
+        <Form.Dropdown.Item value="originalImage" title="Original Image" />
+        <Form.Dropdown.Item value="croppedSubject" title="Cutout" />
+      </Form.Dropdown>
+
+      <Form.TextField
+        title="Width x Height"
+        placeholder="1000x1000"
+        info="The width and height of the output image in pixels. Ignored if output size is not custom."
+        {...itemProps.widthxHeight}
+      />
+
+      <Form.TextField
+        title="Max Height"
```

Comment:
What would be the difference in setting max height/width and the widthxheight field? Also, I was able to insert non-numbers charatcers on that field. That can be avoided.
Maybe for the widthxheight field, we can have a few pre-defined options?

![CleanShot 2024-07-05 at 15 13 50](https://github.com/raycast/extensions/assets/611002/6e3cb751-e7ec-47ad-b705-3a8df867475c)

Reply:
The difference between "Width x Height" and "Max Height/Width" is described when hovering over the info icons to the right of the text input fields. It essentially depends on the selected "Output Size".

That's also why you were able to put in text into "Max Height" while having the "Custom" output size selected. When "Custom" is selected, "Max Height" and "Max Width" are ignored.

Reply:
Pre-defined options for sizes are a good idea, but non-essential for a first release, I'd say.